lib/fa-harness-tools/github_client.rb

```diff
@@ -91,14 +91,20 @@ def get_commit_sha_from_tag(tag)
     #
     # @return [Bool] True is <ancestor> is ancestor of <commit>
     def is_ancestor_of?(ancestor, commit)
-      !!find_commit(commit) { |c| c[:sha] == ancestor }
+      # Compare returns the merge base, the common point in history between the
+      # two commits. If X is the ancestor of Y, then the merge base must be X.
+      # If not, it's a different branch.
+      @octokit.compare(owner_repo, ancestor, commit)[:merge_base_commit][:sha] == get_commit_sha(ancestor)
```

Comment:
Does this work when we are deploying the last known good SHA, with the assumption that every commit we are deploying is a merge commit?

Reply:
Do you mean if `ancestor` and `commit` are the same, will it work (yes, it will)? Or something else?